aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/S3ClientOptions.java

```diff
@@ -55,6 +62,22 @@ public boolean isPathStyleAccess() {
 
     /**
      * <p>
+     * Returns whether the client uses chunked encoding for all requests.
+     * </p>
+     * <p>
+     * The default behaviour is to detect API request which enables chunked
+     * encoding automatically for PutObjectRequest and UploadPartRequest.
+     * Setting this flag will result in disabling chunked encoding for all
+     * requests.
+     * </p>
+     * @return True is the client should always enable chunked encoding.
+     */
+    public boolean isChunkedEncoding() {
```

Comment:
Thinking more about this, it might make sense to expose the property just as disableChunkedEncoding. I think it may be confusing if a developer calls setUseChunkedEncoding(true) and we only don't always do chunked encoding (i.e. we only do chunked encoding for PutObjectRequest or UploadPartRequest). It's also less clear that the default is enabled.

So how about 

isChunkedEncodingDisabled()
disableChunkedEncoding()